src/pip/_internal/utils/temp_dir.py

```diff
@@ -196,7 +196,11 @@ def onerror(
                 traceback.format_exception_only(type(exc_val), exc_val)
             )
             formatted_exc = formatted_exc.rstrip()  # remove trailing new line
-            if func in (os.unlink, os.remove, os.rmdir):
+            if func in (
+                os.unlink,
+                os.remove,
+                os.rmdir,
+            ):  # type: ignore[comparison-overlap]
```

Comment:
This is the only annoying false positive

Reply:
What is causing the false positive? I’m wondering if we should rewrite this or annotate the code better instead.

Reply:
If you annotate with Callable, mypy won't complain. I've pushed this change to the PR.

(I generally don't recommend use of `types.FunctionType` as an annotation, but in this instance it seems fine and mypy should have accepted the code as is)